src/traces/carpet/defaults.js

```diff
@@ -21,6 +21,8 @@ module.exports = function supplyDefaults(traceIn, traceOut, dfltColor, fullLayou
         return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
     }
 
+    traceOut._clipPathId = 'clip' + traceOut.uid + 'carpet';
```

Comment:
@rreusser FYI - it turns out that we weren't applying the clip path to mocks that have the data trace before the carpet trace because the ID was cached during the plot step. I moved `_clipPathId` up to `supplyDefaults` to get around that. It's barely visible in the test images, but that's why they all changed (all the ones that put the data before the carpet anyway).

Also two more changes in here:
- `_clipPathId` rather than `clipPathId` - if we're attaching something to the trace (or layout) that's **not** an attribute, it should either be a function or it should start with an underscore.
- use `Drawing.setClipUrl` rather than directly setting the `clip-path` attribute. This is to handle pages with a `<base>` tag, which is obnoxious but important if it's present - which happens most notably in angular 😑